infer/lib/python/inferlib/capture/gradle.py

```diff
@@ -31,9 +30,136 @@ def gen_instance(*args):
 create_argparser = util.base_argparser(MODULE_DESCRIPTION, MODULE_NAME)
 
 
+def extract_filepath(parts):
+    pos = 0
```

Comment:
Simply reversing the order of the iteration here solves the perf regression issue. No need to update your PR (unless you want to!), I can do the change on my end.

```Python
def extract_filepath(parts):
    size = len(parts)
    pos = size - 1
    while pos >= 0:
        path = ' '.join(itertools.islice(parts, pos, None))
        if os.path.isfile(path):
            return parts[:pos], path
        pos -= 1
    return parts, None
```